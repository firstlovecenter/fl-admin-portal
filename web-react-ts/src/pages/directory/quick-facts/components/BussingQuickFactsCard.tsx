import React from 'react'
import { Badge } from 'react-bootstrap'
import '../QuickFacts.css'
import { getPercentageChange } from './quick-fact-utils'

export interface BussingQuickFactsProps {
  bussingDetails: {
    churchType: string
    cardType: string
    leadersName: string
    churchName: string
    churchBussingThisMonth: number | string
    avgHigherLevelBussingThisMonth: number | string
    higherLevelName: string
  }[]
}

const BussingQuickFactsCard = (props: BussingQuickFactsProps) => {
  const details = props?.bussingDetails[0]

  const percentageRiseOrFall = getPercentageChange(
    details?.churchBussingThisMonth as number,
    details?.avgHigherLevelBussingThisMonth as number
  )

  const getBadgeBackground = () => {
    if (percentageRiseOrFall >= 0) return 'green'
    return 'red'
  }

  const getBadgeColor = () => {
    if (percentageRiseOrFall >= 0) return 'badge-percentage-green'
    return 'badge-percentage-red'
  }

  return (
    <div className="w-100 text-center quick-fact-card">
      <div className="church-text">{details?.churchType}</div>
      <div className="stat-text "> Average {details?.cardType} </div>
      <div className="leader-text">{details?.leadersName}</div>
      <div className="branch-text">
        {details?.churchName + ' ' + details?.churchType}
      </div>
      <div className="bussing-number">
        {details?.churchBussingThisMonth === 'null'
          ? '--'
          : details?.churchBussingThisMonth}
      </div>
      <div>
        <Badge
          bg={`${getBadgeBackground()}`}
          className={`${getBadgeColor()} mt-auto`}
        >
          {percentageRiseOrFall >= 0 ? '+' : ''}
          {percentageRiseOrFall}%
        </Badge>
      </div>
      <hr className="separator" />
      <div className="bussing-number">
        {details?.avgHigherLevelBussingThisMonth === 'null'
          ? '--'
          : details?.avgHigherLevelBussingThisMonth}
      </div>
      <div className="average-text">
        Average {details?.churchType} <br /> Bussing
      </div>
      <div className="higher-church-text">{details?.higherLevelName}</div>
    </div>
  )
}

export default BussingQuickFactsCard
