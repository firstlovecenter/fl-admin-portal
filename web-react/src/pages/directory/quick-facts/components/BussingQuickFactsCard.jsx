import React from 'react'
import { Badge } from 'react-bootstrap'
import '../QuickFacts.css'

const BussingQuickFactsCard = (props) => {
  const details = props?.bussingDetails[0]

  const getPercentageChange = (avgBussing, avgHigherLevelBussing) => {
    var diff = avgBussing - avgHigherLevelBussing
    if (isNaN(diff)) return '--'
    return Math.round((diff / avgBussing) * 100)
  }

  const percentageRiseOrFall = getPercentageChange(
    details?.churchBussingThisMonth,
    details?.avgHigherLevelBussingThisMonth
  )

  const getBadgeBackground = () => {
    if (percentageRiseOrFall >= 0) return 'green'
    return 'red'
  }

  const getBadgeColor = () => {
    if (percentageRiseOrFall >= 0) return 'badge-percentage-green'
    return 'badge-percentage-red'
  }

  return (
    <div className="w-100 text-center quick-fact-card">
      <div className="church-text">{details?.churchType}</div>
      <div className="stat-text "> Average {details?.cardType} </div>
      <div className="leader-text">{details?.leadersName}</div>
      <div className="branch-text">
        {details?.churchName + ' ' + details?.churchType}
      </div>
      <div className="bussing-number">
        {details?.churchBussingThisMonth === 'null'
          ? '--'
          : details?.churchBussingThisMonth}
      </div>
      <div>
        <Badge
          bg={`${getBadgeBackground()}`}
          className={`${getBadgeColor()} mt-auto`}
        >
          {percentageRiseOrFall >= 0 ? '+' : ''}
          {percentageRiseOrFall}%
        </Badge>
      </div>
      <hr className="separator" />
      <div className="bussing-number">
        {details?.avgHigherLevelBussingThisMonth === 'null'
          ? '--'
          : details?.avgHigherLevelBussingThisMonth}
      </div>
      <div className="average-text">
        Average {details?.churchType} <br /> Bussing
      </div>
      <div className="higher-church-text">{details?.higherLevelName}</div>
    </div>
  )
}

export default BussingQuickFactsCard
