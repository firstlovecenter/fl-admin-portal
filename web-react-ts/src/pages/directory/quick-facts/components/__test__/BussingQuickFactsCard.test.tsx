import { render, cleanup } from '@testing-library/react'
import React from 'react'
import BussingQuickFactsCard from '../BussingQuickFactsCard'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

const bussingDetails = [
  {
    churchType: 'Bacenta',
    cardType: 'Bussing',
    leadersName: 'Edem Ahadzi',
    churchName: 'Madina Bacenta',
    churchBussingThisMonth: '23',
    avgHigherLevelBussingThisMonth: '63',
    higherLevelName: 'Madina Council',
  },
]

afterEach(cleanup)

it('renders without crashing', () => {
  render(<BussingQuickFactsCard bussingDetails={bussingDetails} />)
})

it('renders Bussing Card with correct details', () => {
  const { getByTestId } = render(
    <BussingQuickFactsCard bussingDetails={bussingDetails} />
  )
  expect(getByTestId('bussingCard')).toHaveTextContent('Bussing')
})

it('matches snapshot', () => {
  const tree = renderer
    .create(<BussingQuickFactsCard bussingDetails={bussingDetails} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
