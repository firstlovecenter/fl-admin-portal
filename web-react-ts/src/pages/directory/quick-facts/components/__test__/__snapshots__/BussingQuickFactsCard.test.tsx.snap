// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="w-100 text-center quick-fact-card"
  data-testid="bussingCard"
>
  <div
    className="church-text"
  >
    Bacenta
  </div>
  <div
    className="stat-text "
  >
     Average 
    Bussing
     
  </div>
  <div
    className="leader-text"
  >
    Edem Ahadzi
  </div>
  <div
    className="branch-text"
  >
    Madina Bacenta Bacenta
  </div>
  <div
    className="bussing-number"
  >
    23
  </div>
  <div>
    <span
      className="badge-percentage-red mt-auto badge bg-red"
    >
      -63
      %
    </span>
  </div>
  <hr
    className="separator"
  />
  <div
    className="bussing-number"
  >
    63
  </div>
  <div
    className="average-text"
  >
    Average 
    Bacenta
     
    <br />
     Bussing
  </div>
  <div
    className="higher-church-text"
  >
    Madina Council
  </div>
</div>
`;
